src/dc_receive_imf.rs

```diff
@@ -1026,44 +1036,22 @@ fn create_or_lookup_group(
             }
         }
     }
-    if !X_MrGrpImageChanged.is_empty() {
-        info!(
-            context,
-            "grp-image-change {} chat {}", X_MrGrpImageChanged, chat_id
-        );
-        let mut changed = false;
-        let mut grpimage: Option<BlobObject> = None;
-        if X_MrGrpImageChanged == "0" {
-            changed = true;
-        } else {
-            for part in &mut mime_parser.parts {
-                if part.typ == Viewtype::Image {
-                    grpimage = part
-                        .param
-                        .get_blob(Param::File, context, true)
-                        .unwrap_or(None);
-                    info!(context, "found image {:?}", grpimage);
-                    changed = true;
+    if mime_parser.group_avatar != AvatarAction::None {
+        info!(context, "group-avatar change for {}", chat_id);
+        if let Ok(mut chat) = Chat::load_from_db(context, chat_id) {
+            match &mime_parser.group_avatar {
+                AvatarAction::Change(profile_image) => {
+                    chat.param.set(Param::ProfileImage, profile_image);
+                    true
                 }
-            }
-        }
-        if changed {
-            info!(
-                context,
-                "New group image set to '{}'.",
-                grpimage
-                    .as_ref()
-                    .map(|blob| blob.as_name().to_string())
-                    .unwrap_or_default()
-            );
-            if let Ok(mut chat) = Chat::load_from_db(context, chat_id) {
-                match grpimage {
-                    Some(blob) => chat.param.set(Param::ProfileImage, blob.as_name()),
-                    None => chat.param.remove(Param::ProfileImage),
-                };
-                chat.update_param(context)?;
-                send_EVENT_CHAT_MODIFIED = true;
-            }
+                AvatarAction::Delete => {
+                    chat.param.remove(Param::ProfileImage);
+                    true
+                }
+                AvatarAction::None => false,
+            };
```

Comment:
again, the true/true/false returns are dropped  immediately (rust only warns about dropped results but maybe clippy could be made to warn about unused bools, not sure)

Reply:
also here, the cases seems to be of the same return type.

Reply:
i was wrong, removing true/false and adding a {} for the none.part does the trick. i learnt something new, great!

@hpk42 shall i push this? or are you o it?